### DISCUSSION TITLE
DataGrid Component issues

### DISCUSSION THREAD

**Discussion by vignesh-sivaprakasam** [twilio-labs/paste#2140] [Q&A] [answered]
Hi team,
      I am currently migrating my code from `Table` component to `DataGrid` component. 
      Currently I am facing 2 problems.
1. `onClick` prop was supported in `Td` whereas it is not available in `DataGridCell`. Is this intentional ?
2. During snapshot testing, `DataGridCell` generates `id` attribute which gets changed on every test run. so I am unable to use `expect(container).toMatchSnapshot();` in testing.

**Comment by kate-porter** [comment]
Hey @vignesh-sivaprakasam - so sorry for the delayed reply on this. I've shared this with the broader team and we will get back to you very shortly!

**Comment by kate-porter** [comment, marked as answer]
Hey @vignesh-sivaprakasam - 

Currently `onClick` is not supported on the cell because it is used for cell focus, for keyboard support. The current suggested approach is for the element within the cell to have the onClick, but if you could share more about what you're trying to achieve, we can discuss other options if needed.

Snapshot testing works best with deterministic renders, and the DataGrid is non-deterministic due to the ID generation. Could you share more about why you're hoping to snapshot test the DataGrid itself? The component is fully tested in Paste, so it may not be necessary for you to test it depending on your use case.

**Reply by andioneto** [reply]
Hello @vignesh-sivaprakasam 👋 

Do you need further assistance with this discussion topic or did @katieporter's answer provide the help you needed?

**Reply by vignesh-sivaprakasam** [reply]
Hi @andipants12 ,
    @katieporter's answer filled the gaps which I had, thanks.
